packages/library-legacy/src/programs/vote.ts

```diff
@@ -96,6 +97,17 @@ export type UpdateValidatorIdentityParams = {
   nodePubkey: PublicKey;
 };
 
+export type CompactUpdateVoteStateParams = {
+  voteAccount: PublicKey,
+  voteAuthority: PublicKey,
+  voteStateUpdate: {
+    lockouts: Lockout[],
+    root: number,
+    hash: PublicKey,
```

Comment:
Unfortunately, the legacy library doesn't have an opaque type for a hash, but it looks like the rest of the package uses `string`. It definitely can't be `PublicKey` here, for a few reasons.

However, when it comes to decoding, you _can_ use the `publicKey` layout, since it will perform base58 decoding and assert the resulting length is 32 bytes.